app/src/main/python/gogdl/api.py

```
@@ -0,0 +1,118 @@
+        try:
+            response = self.get_authenticated_request(url)
+            
+            if response.status_code != 200:
+                self.logger.warning(f"Invalid secure link response: {response.status_code}")
+                time.sleep(0.2)
+                return self.get_secure_link(product_id, path, generation, root)
+            
+            js = response.json()
+            return js.get('urls', [])
+            
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)